components/user_settings/notifications/email_notification_setting/email_notification_setting.tsx

```diff
@@ -94,12 +103,12 @@ export default class EmailNotificationSetting extends React.PureComponent {
         return null;
     }
 
-    handleChange = (e) => {
-        const enableEmail = e.currentTarget.getAttribute('data-enable-email');
+    handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const enableEmail = JSON.parse(e.currentTarget.getAttribute('data-enable-email')!);
```

Comment:
@hiendinhngoc is it possible that adding this `JSON.parse` statement could be causing the error seen by @DHaussermann? I checked the PR SpinWick and can also reproduce it there.

Reply:
@devinbinnie I've checked `enableEmail` variable and it returns the right value base on user selection. So, I don't think this is the cause.

Reply:
@hiendinhngoc See this line: https://github.com/mattermost/mattermost-webapp/blob/5da1daa84bdbf58eec4c7872091d6ead01a7d457/components/user_settings/notifications/user_settings_notifications.jsx#L942

The parent component is expecting that `enableEmail` is a string, so even when it is set to `true`, it would evaluate to `false` due to the explicit type checking. When `handleChange` is called, it makes a call to a parent method `onChange` that sets the state of the parent component to the pure boolean value, which we don't want. We'll want to change this back to use a `string` for now until we migrate the parent component at least.

Reply:
@devinbinnie thank you so much for your suggestion, I'm going to update it.